pinot-core/src/main/java/org/apache/pinot/core/common/datatable/DataTableImplV3.java

```diff
@@ -350,32 +340,29 @@ private byte[] serializeMetadata()
    * This is to make V3 implementation keep the consumers of Map<String, String> getMetadata() API in the code happy
    * by internally converting it.
    */
-  private Map<String, String> deserializeMetadata(byte[] bytes)
+  private Map<String, String> deserializeMetadata(ByteBuffer buffer)
```

Comment:
Can we add a comment / note stating that the caller of this method assumes that ByteBuffer position is moved accordingly by reads done in this method and so we should use relative get/read methods ?